test/hpke_test.c

```diff
@@ -1212,37 +1256,206 @@ static int test_hpke_oddcalls(void)
     size_t cipherlen = sizeof(cipher);
     unsigned char clear[OSSL_HPKE_TSTSIZE];
     size_t clearlen = sizeof(clear);
+    unsigned char fake_ikm[OSSL_HPKE_TSTSIZE];
+    OSSL_LIB_CTX *badctx = NULL;
+    char *badpropq = "yeah, this won't work";
+    uint64_t lseq = 0;
+
+    /* 
+     * a bunch of these generated specifically to get batter coverage 
+     * (according to lcov), so they're a bit all over the place
+     */
+
+ 
+
+    /* NULL ctx calls */
+    OSSL_HPKE_CTX_free(NULL);
+    if (!TEST_false(OSSL_HPKE_CTX_set_seq(NULL, 1)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_get_seq(NULL, &lseq)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_authpub(NULL, pub, publen)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_authpriv(NULL, privp)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_ikme(NULL, NULL, 0)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_CTX_set1_psk(NULL, NULL, NULL, 0)))
+        goto end;
+
+    /* make/breack ctx */
+    if (!TEST_ptr(ctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                          testctx, "foo")))
+        goto end;
+    OSSL_HPKE_CTX_free(ctx);
 
+    /* bad suite calls */
+    hpke_suite.aead_id = 0xbad;
+    if (!TEST_false(OSSL_HPKE_suite_check(hpke_suite)))
+        goto end;
+    hpke_suite.aead_id = OSSL_HPKE_AEAD_ID_AES_GCM_128;
+    if (!TEST_false(OSSL_HPKE_suite_check(bad_suite)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_get_recommended_ikmelen(bad_suite)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_get_public_encap_size(bad_suite)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_get_ciphertext_size(bad_suite, 0)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_keygen(bad_suite, pub, &publen, &privp,
+                                     NULL, 0, testctx, badpropq)))
+        goto end;
+    if (!TEST_false(OSSL_HPKE_keygen(bad_suite, pub, &publen, &privp,
+                                     NULL, 0, testctx, NULL)))
+        goto end;
+
+    /* dodgy libctx/propq */
+    badctx = (OSSL_LIB_CTX *) 0xff;
```

Comment:
not sure this is a good idea? Did you run valgrind?

Reply:
yep, it seems happy